src/hooks/useCreateGenerator.test.ts

```diff
@@ -0,0 +1,92 @@
+import { useNavigate } from 'react-router-dom'
+import { useCreateGenerator } from './useCreateGenerator'
+import { createNewGeneratorFile } from '@/utils/generator'
+import { generateFileNameWithTimestamp } from '@/utils/file'
+import { getRoutePath } from '@/routeMap'
+import { useToast } from '@/store/ui/useToast'
+import { beforeEach, describe, expect, it, vi, Mock } from 'vitest'
+import { renderHook } from '@testing-library/react'
+import { act } from 'react'
+
+vi.mock('react-router-dom', () => ({
+  useNavigate: vi.fn(),
+}))
+vi.mock('@/utils/generator', () => ({
+  createNewGeneratorFile: vi.fn(),
+}))
+vi.mock('@/utils/file', () => ({
+  generateFileNameWithTimestamp: vi.fn(),
+}))
+vi.mock('@/routeMap', () => ({
+  getRoutePath: vi.fn(),
+}))
+vi.mock('@/store/ui/useToast', () => ({
+  useToast: vi.fn(),
+}))
+
+describe('useCreateGenerator', () => {
+  const navigate = vi.fn()
+  const showToast = vi.fn()
+
+  beforeEach(() => {
+    ;(useNavigate as Mock).mockReturnValue(navigate)
+    ;(useToast as Mock).mockReturnValue(showToast)
+    vi.clearAllMocks()
+  })
+
+  it('should navigate to the correct path on successful generator creation', async () => {
+    const newGenerator = { id: 'test' }
+    const fileName = 'test-file.json'
+    const routePath = '/generator/test-file.json'
+
+    ;(createNewGeneratorFile as Mock).mockReturnValue(newGenerator)
+    ;(generateFileNameWithTimestamp as Mock).mockReturnValue(fileName)
+    ;(getRoutePath as Mock).mockReturnValue(routePath)
```

Comment:
By casting `as Mock` you lose type safety and `mockReturnValue` accepts `any`.
There's a helper `vi.mocked` which allows removing cast (as well as this ugly syntax 😅) and preserve type safety:

```suggestion
    vi.mocked(createNewGeneratorFile).mockReturnValue(newGenerator)
    vi.mocked(generateFileNameWithTimestamp).mockReturnValue(fileName)
    vi.mocked(getRoutePath).mockReturnValue(routePath)
```

You'll get a typescript error on the first line because `{ id: 'test' }` is not a valid `GeneratorFileData`, but that's expected, and a fixture should be created